File: Backend/ApiUniversidad/ApiUniversidad/Models/Docente.cs
using System;
using System.Collections.Generic;

namespace ApiUniversidad.Models;

public partial class Docente
{
    public Guid Id { get; set; }

    public string Nombre { get; set; } = null!;

    public string Apellido { get; set; } = null!;

    public string Legajo { get; set; } = null!;

    public Guid IdRol { get; set; }

    public DateTime FechaAlta { get; set; }

    public virtual ICollection<DocentesPorCurso> DocentesPorCursos { get; set; } = new List<DocentesPorCurso>();

    public virtual Role IdRolNavigation { get; set; } = null!;
}

File: Frontend/ApiUniversidad/UniversidadAngular/src/app/models/docente.ts
import { DocentesPorCurso } from "./docentes-por-curso";
import { Rol } from "./rol";

export class Docente {

    id?: string;
    nombre: string = '';
    apellido: string = '';
    legajo: string = '';
    idRol: string = '';
    fechaAlta: Date = new Date();
    docentesPorCurso: DocentesPorCurso[] = [];
    idRolNavigation: Rol = new Rol();

}
